jmolecules-events/src/main/java/org/jmolecules/event/annotation/DomainEventPublisher.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jmolecules.event.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Identifies a domain event publisher, i.e. logic to publish a {@link DomainEvent}.
```

Comment:
It is unclear where this annotation is expected to be placed. On the Command handler method that is expected to publish an event as part of its processing? (As described in my comments on CommandPublisher)

Similarly, I don't see much value add in the `publishes` attribute on this annotation. The event being published may vary at runtime and should always be passed as a parameter anyway. The type mentioned on that parameter is the actual type being published.

Reply:
That's kind of in line with what I tried to argue [here](https://github.com/xmolecules/jmolecules/pull/39/files/e9f616fb3f18d51b63f0880de924365af94fc3e4#r545101987)

Reply:
see: https://github.com/xmolecules/jmolecules/pull/39#discussion_r545940451